src/main/java/seedu/duke/parsers/AddParser.java

```diff
@@ -12,85 +12,67 @@
-    * (?=(\s+-d\s+)|$) -- asserts -d might follow
-    * (\s+(-d\s+\"(?<taskDescription2>([^\"]*))\"))? -- captures taskDescription2 with same constraints 
-    *                                                   if -t precedes -d
-    * \s*(( ... |\/m\s+(?<moduleCode>\w+? ... ) -- alternatively captures moduleCode
-    *                                              (no whitespaces or special characters)
-    * (?=(\s+-d\s+)|\s+|$) -- asserts -d or whitespaces (if command is invalid) might follow
-    * (\s+(-d\s+\"(?<moduleDescription>([^\"]*))\"))? -- captures moduleDescription (cannot have ")
-    *                                                    which must be enclosed with "". Optional
-    * \s+( Task | Module )(?<invalid>.*) -- any input at any part of the command
-    *                                       that does not fit the pattern will be captured as invalid
-    * */
-    // TODO: Add support for -mod argument when integrating Task and Module classes with one another
-    private static final String ADD_FORMAT = "\\s*((\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))"
-            + "(\\s+(-d\\s+\\\"(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
-            + "(?<estimatedWorkingTime>([^\\\"]*))\\\")(?=(\\s+-d\\s+)|$))?(\\s+(-d\\s+\\\""
-            + "(?<taskDescription2>([^\\\"]*))\\\"))?|\\/m\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|\\s+|$))(\\s+(-d\\s+"
-            + "\\\"(?<moduleDescription>([^\\\"]*))\\\"))?))(?<invalid>.*)";
+    /* Explanation for regex:
+     * (/t\s+\"(?<taskName>[^\"]+)\"                     -- matches [/t "taskName"].
+     * (\\s+-d\\s+\\\"(?<taskDescription>[^\\\"]+)\\\")? -- matches [-d "taskDescription"] if present. Optional
+     * (\s+-t\s+\"(?<estimatedWorkingTime>[^\"]+)\")?    -- matches [-t "estimatedWorkingTime"] if present. Optional
+     *                                                   -- None of the above fields accept " as a valid character.
+     *
+     * (\s+-m\s+\"(?<taskModule>\w+)\")?                 -- matches [-m taskModule] if present. Optional
+     *                                                      Note that taskModule does not require "", but must be a
+     *                                                      single word composed of [a-zA-Z0-9_].
+     *
+     * /m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|$))          -- matches [/m moduleCode]
+     *                                                      Same as above, note that moduleCode does not require "",
+     *                                                      but must also be a single word composed of [a-zA-Z0-9_].
+     *
+     * (\s+(-d\s+\"(?<moduleDescription>.+)\"))?)        -- matches [-d "moduleDescription"] if present. Optional
+     *                                                      Does not accept " as a valid character.
+     */
+
+    private static final String ADD_FORMAT = "(/t\\s+\\\"(?<taskName>[^\\\"]+)\\\"(\\s+-d\\s+\\\""
```

Comment:
i think keeping the invalid group can make the input stricter which could reduce potential bugs

Reply:
Invalid group is redundant so no changes are needed